doc/lsst.pipe.base/creating-a-pipeline.rst

```diff
@@ -0,0 +1,448 @@
+    class: lsst.pipe.tasks.makeCoaddTempExp.MakeWarpTask
+    config:
+      matchingKernelSize: 29
+      makePsfMatched: true
+      modelPsf.defaultFwhm: 7.7
+      doApplyExternalPhotoCalib: false
+      doApplyExternalSkyWcs: false
+      doApplySkyCorr: false
+      doWriteEmptyWarps: true
+
+This example shows an entry for
+`~lsst.pipe.tasks.makeCoaddTempExp.MakeWarpTask`. The label used for this task 
+is ``makeWap`` and the class location is now declared in the sub mapping
+alongside the ``class`` keyword. The ``config`` keyword is associated with 
+various `~lsst.pex.config.Field`\ s and the configuration appropriate for this 
+`Pipeline` specified as an additional yaml mapping.
+
+The complete complexity of `lsst.pex.config` can't be represented with simple
+yaml mapping syntax. To account for this, ``config`` blocks in `Pipeline`\ s
+support two special fields: ``file`` and ``python``.
+
+The ``file`` key may be associated with either a single value pointing to a
+filesystem path where a `lsst.pex.config` file can be found, or a yaml list
+of such paths. The file paths can contain environment variables that will be
+expanded prior to loading the file(s). These files will then be applied to
+the task during configuration time to override any default values.
+
+Sometimes configuration is too complex to express with yaml syntax, yet it is
+simple enough that it does not warrant its own config file. The ``python``
+key is designed to support this use case. The value associated with the key
+is a (possibly multi-line) string with valid python syntax. This string is
+evaluated and applied during task configuration exactly as if it had been
+written in a file or typed out in an interpreter. The following example expands
+the previous one to use the ``python`` key.
+
+.. code-block:: yaml
+
+  makeWarp:
+    class: lsst.pipe.tasks.makeCoaddTempExp.MakeWarpTask
+    config:
+      matchingKernelSize: 29
+      makePsfMatched: true
+      modelPsf.defaultFwhm: 7.7
+      doApplyExternalPhotoCalib: false
+      doApplyExternalSkyWcs: false
+      doApplySkyCorr: false
+      doWriteEmptyWarps: true
+      python: "config.warpAndPsfMatch.psfMatch.kernel['AL'].alardSigGauss = \
+        [1.0, 2.0, 4.5]"
+
+----------
+Parameters
+----------
+
+As you saw in the pervious section, each task defined in a `Pipeline` may
+have its own configuration. However, it is sometimes useful for configuration
+fields in multiple tasks to share the same value. `Pipeline`\ s support this
+with a concept called ``parameters``. This is a top level section in the
+`Pipeline` document specified with a key named ``parameters``.
+
+The contents of the ``parameters`` section is a mapping of key, value pair-
+associated value in the parameters section.
+
+To make this a bit clearer take a look at the following example, making note
+that only config fields relevant for this example are shown.
+
+.. code-block:: yaml
+
+  parameters:
+    calibratedSingleFrame: calexp
```

Comment:
I updated to reflect that currently it can only be used in the key-value sections of config overrides. It can take any valid value, bool, float, etc. I tried to make that clearer in the doc as well.